pkg/controller/errors/errors.go

```diff
@@ -47,6 +47,21 @@ func IsMultipleExistingCRDOwnersError(err error) bool {
 	return false
 }
 
+type RetryableError struct {
+	error
+}
+
+func NewRetryableError(err error) RetryableError {
+	return RetryableError{err}
+}
+func IsRetryable(err error) bool {
+	switch err.(type) {
+	case RetryableError:
+		return true
+	}
+	return false
+}
```

Comment:
Should we investigate an [`errors.Is(...)]`(https://pkg.go.dev/errors#example-Is) implementation? I'm not sure we need this kind of wrapper logic since that method was introduced in Go 1.13.

Reply:
I think the problem is we don't really return any typed errors in the code. Just plain old Error. So, it becomes difficult to discern the retryability of errors. To use straight up `errors.Is(...)`, I think we'd need to create some sort of `CSVError` and add some `retryable` boolean to that. The nice thing about the wrapper is that it can wrap any error as a retryable.

Reply:
So leave this as is?